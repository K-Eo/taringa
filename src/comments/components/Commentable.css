.Commentable {
  display: flex;
  margin-bottom: 1rem;
}

.Commentable .btn-light {
  font-size: 13px;
}

.Commentable:last-child {
  margin-bottom: 0 !important;
}

.Commentable-avatar {
  max-width: 32px;
  max-height: 32px;
  min-width: 32px;
  min-height: 32px;
}

.Commentable-author {
  margin-bottom: 0;
  font-size: 90%;
  font-weight: bold;
}

@media (min-width: 992px) {
  .Commentable-avatar {
    max-width: 40px;
    max-height: 40px;
    min-width: 40px;
    min-height: 40px;
  }
}

.Commentable-meta-created {
  color: #a6a6a6;
  font-weight: normal;
  margin-left: 0.5rem;
}

.Commentable-body img {
  max-width: 100%;
}

.Commentable iframe {
  max-width: 100%;
}

.Commentable-body {
  font-size: 95%;
  word-break: break-word;
}

.Commentable-body span[style^='position:relative'] {
  overflow: hidden;
  height: 16px;
}

.Commentable-replies-toggler {
  padding: 0;
  font-weight: bold;
  text-decoration: none;
  font-size: 95%;
}

.Commentable-replies-toggler .fa {
  margin-left: 4px;
}

.Commentable-replies-toggler:hover,
.Commentable-replies-toggler:focus,
.Commentable-replies-toggler:active {
  text-decoration: none;
}
