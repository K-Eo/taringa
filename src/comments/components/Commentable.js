import React, { PureComponent } from 'react'
import TimeAgo from 'react-timeago'

import './Commentable.css'
import Action from '../../common/Action.js'
import Avatar from '../../users/components/Avatar'
import UsernameLink from '../../users/components/UsernameLink'

class Commentable extends PureComponent {
  render() {
    const { body, children, created, downvotes, owner, upvotes } = this.props

    return (
      <div className="Commentable">
        <div className="Commentable-avatar mr-2 mr-lg-3">
          <Avatar id={owner} />
        </div>
        <div className="Commentable-content">
          <p className="Commentable-author">
            <UsernameLink className="text-dark" id={owner} />

            <TimeAgo
              className="Commentable-meta-created"
              date={created}
              formatter={Intl.ES()}
            />
          </p>

          <div
            className="Commentable-body"
            dangerouslySetInnerHTML={{ __html: body }}
          />

          <p className="my-1">
            <Action className="mr-3" icon="far fa-thumbs-up">
              {upvotes.humanize()}
            </Action>

            <Action className="mr-3" icon="far fa-thumbs-down">
              {downvotes.humanize()}
            </Action>

            <Action className="d-none d-sm-inline">RESPONDER</Action>
            <Action className="d-sm-none" icon="far fa-comment-alt" />
          </p>

          {children}
        </div>
      </div>
    )
  }
}

Commentable.defaultProps = {
  body: '#BODY',
  created: 0,
  downvotes: 0,
  owner: '#OWNER',
  upvotes: 0,
}

export default Commentable
