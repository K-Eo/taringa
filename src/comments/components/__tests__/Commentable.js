import React from 'react'

import Commentable from '../Commentable'

describe('Commentable', () => {
  it('exists', () => {
    expect(Commentable).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<Commentable />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders with given comment', () => {
    const comment = {
      body: 'foo',
      create: 1513924592705,
      downvotes: 2,
      upvotes: 5,
    }

    const wrapper = shallow(<Commentable comment={comment} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders with user', () => {
    const owner = {
      avatar: 'bar',
      username: 'foo',
    }

    const wrapper = shallow(<Commentable owner={owner} />)

    expect(wrapper).toMatchSnapshot()
  })
})
