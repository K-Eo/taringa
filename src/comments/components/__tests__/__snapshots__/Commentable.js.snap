// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commentable renders default 1`] = `
<div
  className="Commentable"
>
  <div
    className="Commentable-avatar mr-2 mr-lg-3"
  >
    <img
      alt="#USERNAME"
      className="rounded Commentable-avatar"
      src="#AVATAR"
    />
  </div>
  <div
    className="Commentable-content"
  >
    <p
      className="Commentable-author"
    >
      <Link
        className="text-dark"
        replace={false}
        to="/u/#USERNAME"
      >
        #USERNAME
      </Link>
      <TimeAgo
        className="Commentable-meta-created"
        component="time"
        formatter={[Function]}
        live={true}
        maxPeriod={Infinity}
        minPeriod={0}
        now={[Function]}
      />
    </p>
    <div
      className="Commentable-body"
      dangerouslySetInnerHTML={
        Object {
          "__html": "#BODY",
        }
      }
    />
    <p
      className="my-1"
    >
      <Action
        className="mr-3"
        icon="far fa-thumbs-up"
      >
        0
      </Action>
      <Action
        className="mr-3"
        icon="far fa-thumbs-down"
      >
        0
      </Action>
      <Action
        className="d-none d-sm-inline"
      >
        RESPONDER
      </Action>
      <Action
        className="d-sm-none"
        icon="far fa-comment-alt"
      />
    </p>
  </div>
</div>
`;

exports[`Commentable renders with given comment 1`] = `
<div
  className="Commentable"
>
  <div
    className="Commentable-avatar mr-2 mr-lg-3"
  >
    <img
      alt="#USERNAME"
      className="rounded Commentable-avatar"
      src="#AVATAR"
    />
  </div>
  <div
    className="Commentable-content"
  >
    <p
      className="Commentable-author"
    >
      <Link
        className="text-dark"
        replace={false}
        to="/u/#USERNAME"
      >
        #USERNAME
      </Link>
      <TimeAgo
        className="Commentable-meta-created"
        component="time"
        formatter={[Function]}
        live={true}
        maxPeriod={Infinity}
        minPeriod={0}
        now={[Function]}
      />
    </p>
    <div
      className="Commentable-body"
      dangerouslySetInnerHTML={
        Object {
          "__html": "foo",
        }
      }
    />
    <p
      className="my-1"
    >
      <Action
        className="mr-3"
        icon="far fa-thumbs-up"
      >
        5
      </Action>
      <Action
        className="mr-3"
        icon="far fa-thumbs-down"
      >
        2
      </Action>
      <Action
        className="d-none d-sm-inline"
      >
        RESPONDER
      </Action>
      <Action
        className="d-sm-none"
        icon="far fa-comment-alt"
      />
    </p>
  </div>
</div>
`;

exports[`Commentable renders with user 1`] = `
<div
  className="Commentable"
>
  <div
    className="Commentable-avatar mr-2 mr-lg-3"
  >
    <img
      alt="foo"
      className="rounded Commentable-avatar"
      src="bar"
    />
  </div>
  <div
    className="Commentable-content"
  >
    <p
      className="Commentable-author"
    >
      <Link
        className="text-dark"
        replace={false}
        to="/u/foo"
      >
        foo
      </Link>
      <TimeAgo
        className="Commentable-meta-created"
        component="time"
        formatter={[Function]}
        live={true}
        maxPeriod={Infinity}
        minPeriod={0}
        now={[Function]}
      />
    </p>
    <div
      className="Commentable-body"
      dangerouslySetInnerHTML={
        Object {
          "__html": "#BODY",
        }
      }
    />
    <p
      className="my-1"
    >
      <Action
        className="mr-3"
        icon="far fa-thumbs-up"
      >
        0
      </Action>
      <Action
        className="mr-3"
        icon="far fa-thumbs-down"
      >
        0
      </Action>
      <Action
        className="d-none d-sm-inline"
      >
        RESPONDER
      </Action>
      <Action
        className="d-sm-none"
        icon="far fa-comment-alt"
      />
    </p>
  </div>
</div>
`;
